File: 2334-subarray-with-elements-greater-than-varying-threshold/2334-subarray-with-elements-greater-than-varying-threshold.js
/**
 * @param {number[]} nums
 * @param {number} threshold
 * @return {number}
 */
var validSubarraySize = function(nums, threshold) {
  nums = [0, ...nums, 0];
  const stack = [0];
  
  for (let i = 1; i < nums.length; i++) {
    while (nums[i] < nums[stack.at(-1)]) {
      const j = stack.pop();
      if (nums[j] > threshold / (i - stack.at(-1) - 1)) 
        return i - stack.at(-1) - 1
    }
    
    stack.push(i);
  }
  
  return -1;
};

File: 2334-subarray-with-elements-greater-than-varying-threshold/2334-subarray-with-elements-greater-than-varying-threshold.py
class Solution:
    def validSubarraySize(self, nums: List[int], threshold: int) -> int:
        nums = [0] + nums + [0]
        stack = [0]
        for i, num in enumerate(nums):
            while num < nums[stack[-1]]:
                j = stack.pop()
                if nums[j] > threshold // (i - stack[-1] - 1):
                    return i - stack[-1] - 1
            stack.append(i)
        return -1